cs411project/views/hbtest_view.py

```diff
@@ -0,0 +1,49 @@
+from flask import g, Flask, request
+from flask.json import jsonify
+from flask.views import MethodView
+from ..database.entity_serializer import EntitySerializer
+
+import json
+
+class HBView(MethodView):
+	def get(self,NetID,MachineID):
+		cursor = g.mysql_connection.get_connection().cursor(prepared=True)
+		#curosr = connection.cursor(prepared=True)
+		#cursor = connectoin.cursor()
+		#query = "SELECT * FROM Machine WHERE MachineID = %s"
+		query = """SELECT NetID,SeqID
+                           FROM HeartbeatSequence
+                           WHERE SeqID IN 
+                               (SELECT Max(SeqID) AS M
+                               FROM HeartbeatSequence
+                               WHERE MachineID = %s
+                               GROUP BY MachineID)
+                               AND
+                               NetID = (%s)
+                               AND
+                               ((LastTS IS NOT NULL AND NOT (LastTS + Tfail < CURRENT_TIMESTAMP))
+                               OR
+                               (LastTS IS NULL AND NOT (FirstTS + Tfail < CURRENT_TIMESTAMP)))"""
+		cursor.execute(query,(MachineID,NetID,))
+		result = list(cursor)
+		#cursor.close()
+		if not result:
```

Comment:
The above query finds the most recent heartbeat sequence with given NetID and MachineID, and it also checks whether it is still within the update time window by checking LastTS+Tfail<CURRENT_TIMESTAMP(if LastTS is not NULL) or FirstTS+Tfail<CURRENT_TIMESTAMP(if LastTS is NULL). If such sequence exists (result is not NULL), we will update the returned sequence, otherwise, we will insert a new sequence.